packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -62,26 +65,55 @@ const styles = StyleSheet.create({
   },
 });
 
-class BpkHorizontalNav extends React.Component {
-  constructor(props) {
-    super(props);
+type ViewProps = ElementProps<typeof View>;
+type ViewStyleProp = $PropertyType<ViewProps, 'style'>;
+
+export type Props = {
+  children: Node,
+  selectedId: string,
+  spaceAround: boolean,
+  style: ViewStyleProp,
+};
+
+type State = {
+  indicatorOffsetX: ?number,
+  indicatorWidth: ?number,
+};
+
+class BpkHorizontalNav extends React.Component<Props, State> {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    selectedId: PropTypes.string.isRequired,
+    spaceAround: PropTypes.bool,
+    style: ViewPropTypes.style,
+  };
+
+  static defaultProps = {
+    spaceAround: false,
+    style: null,
+  };
+
+  constructor() {
+    super();
+
     this.state = {
       indicatorOffsetX: null,
       indicatorWidth: null,
     };
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps(nextProps: Props) {
     if (nextProps.selectedId && this.childrenPositions[nextProps.selectedId]) {
       const nextLayoutProps = this.childrenPositions[nextProps.selectedId];
+
       this.setState({
         indicatorOffsetX: nextLayoutProps.x,
         indicatorWidth: nextLayoutProps.width,
       });
     }
   }
 
-  onChildLayout = (event, id) => {
+  onChildLayout = (event: any, id: string) => {
```

Comment:
I'll give it a try.

Reply:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js:117:44

Cannot get event.nativeEvent.layout because property layout is missing in Event [1].

     packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js
     114│   }
     115│
     116│   onChildLayout = (event: SyntheticAnimationEvent<>, id: string) => {
     117│     const { width, x } = event.nativeEvent.layout;
     118│     this.childrenPositions[id] = { width, x };
     119│
     120│     // If the child in question is the initially selected one, the indicator can now be positioned.

     /private/tmp/flow/flowlib_26c5c686/react-dom.js
 [1] 125│   nativeEvent: Event;
```

Reply:
🤷‍♂️ I'm sure there is a type out there to handle this but I'm not gonna block it because of this.